### ISSUE TITLE
Feathers v3 (Buzzard)

### ISSUE THREAD

**Issue by DaniSchenk** [feathersjs-ecosystem/feathers-chat-angular#7]
Would be great, if someone could upgrade feathers to v3.

**Comment by daffl** [comment]
By all means. [Client upgrades](https://docs.feathersjs.com/migrating.html#updating-client-side-applications) are even easier than on the server, in this case changing https://github.com/feathersjs-ecosystem/feathers-chat-angular/blob/master/src/app/services/feathers.service.ts#L3 from

```js
import * as feathers from 'feathers/client';
```

To

```js
import feathers from '@feathersjs/client';
```

Removing the `hooks` import line from https://github.com/feathersjs-ecosystem/feathers-chat-angular/blob/master/src/app/services/feathers.service.ts#L24 and `feathers-client` from the `package.json` and running `npm install @feathersjs/client --save` should do it.

Also it should probably should just use all of `@feathersjs/client` so https://github.com/feathersjs-ecosystem/feathers-chat-angular/blob/master/src/app/services/feathers.service.ts would look like this:

```ts
import { Injectable } from '@angular/core';

import feathers from '@feathersjs/client';
import * as feathersRx from 'feathers-reactive';
import * as io from 'socket.io-client';

/**
 * Simple wrapper for feathers
 */
@Injectable()
export class Feathers {
  // There are no proper typings available for feathers, due to its plugin-heavy nature
  private _feathers: any;
  private _socket: any;

  constructor() {
    this._socket = io('http://localhost:3030');       // init socket.io
    this._feathers = feathers();                      // init Feathers
    this._feathers.configure(feathersRx({             // add feathers-reactive plugin
      idField: '_id'
    }));
    this._feathers.configure(feathers.socketio(this._socket)); // add socket.io plugin
    this._feathers.configure(feathers.authentication({         // add authentication plugin
      storage: window.localStorage
    }));
  }

  // expose services
  public service(name: string) {
    return this._feathers.service(name);
  }

  // expose authentication
  public authenticate(credentials?): Promise<any> {
    return this._feathers.authenticate(credentials);
  }

  // expose logout
  public logout() {
    return this._feathers.logout();
  }
}
```

**Comment by DaniSchenk** [comment]
Sorry, if I sounded demanding. I just wanted to leave a hint...

**Comment by daffl** [comment]
All good, I was suggesting you could submit a pull request 😄

**Comment by j2L4e** [comment]
closed via #9
